### PR TITLE
fix: ensure gengodep uses vendor dir if present, from sylabs 468

### PR DESCRIPTION
**Pull request by edytuk**
This pulls in sylabs PR

- sylabs/singularity#468
which fixed
- sylabs/singularity#461

The original PR description was:

> In a previous PR, initialization of the `srcdir` variable was removed from `makeit/gengodep`. However, the `gengodep` script was still using `srcdir` when checking for a `${srcdir}/vendor/modules.txt`.
> 
> Ref: [48e7fab](https://github.com/sylabs/singularity/commit/48e7fab83e11a698adbd506d60e13fb375b00fec)
> 
> As a result, the vendor dir is never found, and `go` was run with `-mod=readonly`, which would lead to downloads.
> 
> Since go 1.14 the presence of a vendor dir infers `-mod=vendor` automatically, so we can just remove the handling here.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -105,6 +105,9 @@ For older changes see the [archived Singularity change log](https://github.com/a
   configuration file being written.
 - `--no-https` now applies to connections made to library services specified
   in `library://<hostname>/...` URIs.
+- Ensure `gengodep` in build uses vendor dir when present.
+- Fix `source` of a script on `PATH` and scoping of environment variables in
+  definition files (via dependency update).
```
Comment (DrDaveD) [inline CHANGELOG.md:110]: I think this extra change is a merge error, I don't see it in the original PR.  Please remove it.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (edytuk) [reply CHANGELOG.md]: I removed it.

**Review by DrDaveD — APPROVED** [review]

**Comment by DrDaveD** [comment]
I approved but still would like @ikaneshiro to review.

**Review by ikaneshiro — APPROVED** [review]

**Comment by ikaneshiro** [comment]
@edytuk Looks like there are conflicts to resolve before merging

**Comment by DrDaveD** [comment]
I resolved the conflict.

**Pull request merged**